### DISCUSSION TITLE
How to set buffer text and highlight

### DISCUSSION THREAD

**Discussion by ahmedkhalf** [MunifTanjim/nui.nvim#40] [Q&A] [answered]
Hello, thank you for making this plugin! I would like to add some text inside the buffer and highlight the text.
How?

**Comment by MunifTanjim** [comment, marked as answer]
nui doesn't have dedicated methods for those.

You can use the neovim functions directly:

- `vim.api.nvim_buf_set_lines`
- `vim.api.nvim_buf_set_text`
- `vim.api.nvim_buf_add_highlight`

You get the buffer number from `popup.bufnr`.

So something like this would work:

```lua
vim.api.nvim_buf_set_lines(popup.bufnr, 0, 1, false, { "Hello World" })
```

**Comment by ahmedkhalf** [comment]
Thank you!
